tap_s3_csv/dialect.py

```diff
@@ -0,0 +1,180 @@
+import re
+import random
+import csv
+import chardet
+import clevercsv
+
+from tap_s3_csv import s3
+
+def detect_tables_dialect(config):
+    # there is only one table in the array
+    for table in config['tables']:
+        # will return all matching files in s3 with given prefix and table name in config
+        s3_files = s3.get_input_files_for_table(config, table)
+
+        for s3_file in s3_files:
+            detect_dialect(config, s3_file, table)
+
+
+def detect_dialect(config, s3_file, table):
```

Comment:
Followed dialect detection code in tap-csv.

https://github.com/symon-ai/tap-csv/blob/749ac1818611dc1bc55a84ccf4abee0e495c686d/tap_csv/__init__.py#L205

https://github.com/symon-ai/tap-csv/blob/749ac1818611dc1bc55a84ccf4abee0e495c686d/tap_csv/__init__.py#L16